### DISCUSSION TITLE
June release 0.95 - Build responsive SQL CRUD apps in seconds (Beta)

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#1804] [Announcements]
The June release significantly improves the developer experience when; building CRUD apps with an SQL database, laying out components, and delivering responsive apps.

### TL; DR
1. Build SQL CRUD apps in seconds
2. A new layout settings bar
3. Preset layouts
4. Responsive navigation layout
5. Improved design experience
6. Confirmation modal for actions

**To learn how to update Budibase to get the new features, scroll to the bottom of this discussion.**

---

### Build SQL CRUD apps in seconds
Previous to this release, SQL was supported but the experience was tedious. We talked with our community, we listened hard and went back to first principles, and the outcome is an entirely new and improved SQL experience. With this release, connecting your MySQL or PostgreSQL database to Budibase will pull your information about your tables so you can benefit from the same functionality as internal tables. Expanding on this, you will now be able to:
- Auto-generate ‘list’, ‘detail’, and ‘new’ screens
- View SQL tables within Budibase
- Auto-generate forms from your SQL tables
- Automations are fully compatible with SQL tables
- Effortlessly filter SQL data using the data provider component
- SQL pagination is now supported
- Basic relationship functionality

---

### A new layout settings bar
You can now customize the layout of your user interface (UI) using the new layout settings bar. The settings bar is activated by clicking on a container within the design screen, and it makes designing UIs easier, faster, and in our opinion, greatly improves user experience. 

---

### Layout presets
Many of the apps created with Budibase follow similar design patterns. With this knowledge, we’ve released a new Section component that contains layout presets. These presets mirror the common design patterns, helping users layout new sections in seconds. 

---

### Responsive navigation layout
The new Navigation Layout replaces the existing Top Navigation Layout, and now comes with two presets; top navigation and side navigation. The side navigation is a welcomed addition and common request from the community, and with this release, it’s just 2 clicks away - and it looks awesome. I’m also happy to announce the new navigation layout is responsive and works seamlessly across mobile, tablet, and desktop. To top it off, the navigation layout now comes with a new portal icon that, when clicked, allows users to return to their portal and log out.

---

### Improved design experience
A lot of work has gone into improving the interactivity of the design screen, including component labels, improved highlighting, and component stabilization. Hopefully, you’ll agree, the design experience is smoother, more informative and enjoyable.

---

### Confirmation modal for actions
With 1 click, add a confirmation modal to critical actions within your apps. This will improve the end-user experience around deleting records and completing other irreversible actions.

---

# How to update

To update Budibase, navigate to the Budibase Home section, and click the Updates link on the side panel. Then, click check for updates.
![Screenshot 2021-06-23 at 08 35 45](https://user-images.githubusercontent.com/49767913/123055637-042acd80-d3fe-11eb-8e92-1368c55e8f98.png)

Thank you to everyone who helped with this release. We will push wider communications out next week.  Please, try the new release and share your feedback, right here, in Budibase Discussions.

**Comment by KernOil** [comment]
Great work Guys!

**Reply by mjashanks** [reply]
let us know what you think of the SQL improvements!

**Reply by Manishkparmar80** [reply]
@joebudi  can you pls. release sample training video to quickly understand following improvement mentioned into your post.

You have mentioned MySQL or PostgreSQL database but i understood that improvement is applicable for SQL Server also. Pls. confirm.  

********************
Build SQL CRUD apps in seconds
Previous to this release, SQL was supported but the experience was tedious. We talked with our community, we listened hard and went back to first principles, and the outcome is an entirely new and improved SQL experience. With this release, connecting your MySQL or PostgreSQL database to Budibase will pull your information about your tables so you can benefit from the same functionality as internal tables. Expanding on this, you will now be able to:

Auto-generate ‘list’, ‘detail’, and ‘new’ screens
View SQL tables within Budibase
Auto-generate forms from your SQL tables
Automations are fully compatible with SQL tables
Effortlessly filter SQL data using the data provider component
SQL pagination is now supported
Basic relationship functionality
***********************

**Reply by shogunpurple** [reply]
@Manishkparmar80 this feature is not yeat available for MS SQL Server, but will be in the coming weeks.

Training video: https://docs.budibase.com/data/data-sources/plus-datasources

**Reply by Manishkparmar80** [reply]
@shogunpurple  ok. pls. try to bring the same for MS SQL Server at the earliest.

**Comment by Fabian1410** [comment]
When is the next release?

**Comment by Manishkparmar80** [comment]
@joebudi  is MS SQL included under CRUD app DB in July release.

**Reply by shogunpurple** [reply]
@Manishkparmar80 no, it has not been added. Please refer to the release notes discussion:

https://github.com/Budibase/budibase/discussions/2165
